### PR TITLE
[RFR] Move <ReferenceInput> display logic to a child component

### PR DESCRIPTION
**Pull request by fzaninotto**
Just like the recent `<ReferenceField>` refactoring, this change aims to remove the display responsibility out of the `<ReferenceInput>` component and to its child.

Before:

```js
export const CommentEdit = (props) => (
    <Edit {...props}>
        <DisabledInput label="id" source="id" />
        <ReferenceInput label="Post" source="post_id" reference="posts" referenceSource="title" />
        <DateInput label="date" source="created_at" />
        <LongTextInput label="body" source="body" />
    </Edit>
);
```

After:

```js
export const CommentEdit = (props) => (
    <Edit {...props}>
        <DisabledInput label="id" source="id" />
        <ReferenceInput label="Post" source="post_id" reference="posts">
            <SelectInput optionText="title" />
        </ReferenceInput>
        <DateInput label="date" source="created_at" />
        <LongTextInput label="body" source="body" />
    </Edit>
);
```

It's more verbose, but also much more powerful. Imagine how we can compose buttons list or options list instead of a select box.

```js
export const CommentEdit = (props) => (
    <Edit {...props}>
        <DisabledInput label="id" source="id" />
        <ReferenceInput label="Post" source="post_id" reference="posts">
            <RadioButtonGroupInput optionText="title" />
        </ReferenceInput>
        <DateInput label="date" source="created_at" />
        <LongTextInput label="body" source="body" />
    </Edit>
);
```

Also, it's more consistent with other `Reference` components: they now *always* require a child component to display their result.

### CODE REVIEW

File: docs/Tutorial.md [hunk 1]
```diff
@@ -155,18 +155,20 @@ export const PostList = (props) => (
...
+
 ## Creation and Edition
 
 An admin interface is usually for more than seeing remote data - it's for editing and creating, too. Admin-on-REST provides `<Create>` and `<Edit>` components for that purpose. We'll add them to the `posts` resource:
 
 ```js
 // in src/posts.js
 import React from 'react';
-import { List, Edit, Create, ReferenceField, TextField, EditButton, DisabledInput, LongTextInput, ReferenceInput, TextInput } from 'admin-on-rest/lib/mui';
+import { List, Edit, Create, ReferenceField, TextField, EditButton, DisabledInput, LongTextInput, ReferenceInput SelectObjectInput, TextInput } from 'admin-on-rest/lib/mui';
```
Comment (Kmaschta) [inline docs/Tutorial.md]: What about split these import into subfolders like `admin-on-rest/lib/mui/crud` and `admin-on-rest/lib/mui/fields`?

**Comment by fzaninotto** [comment]
Back to WIP, I'm not satisfied with the API

File: docs/Tutorial.md (hunk 1 above)
Comment (fzaninotto) [inline docs/Tutorial.md]: You can already import them individually. It's just more convenient to require whatever you need from a single file.

**Comment by fzaninotto** [comment]
And back to RFR

**Comment by djhi** [comment]
Much better !

**Pull request merged**